nvme_metrics.py

```diff
@@ -159,75 +166,73 @@ def main():
-                    # discard resolution if converted to a JSON number (i.e., float64_t). Instead,
-                    # nvme-cli marshals them as strings. As such, they need to be explicitly cast
-                    # to int or float when using them in Counter metrics.
-                    metrics["data_units_read"].labels(device_name).inc(
-                        int(smart_log["data_units_read"])
-                    )
-                    metrics["data_units_written"].labels(device_name).inc(
-                        int(smart_log["data_units_written"])
-                    )
-                    metrics["host_read_commands"].labels(device_name).inc(
-                        int(smart_log["host_read_commands"])
-                    )
-                    metrics["host_write_commands"].labels(device_name).inc(
-                        int(smart_log["host_write_commands"])
-                    )
-                    metrics["avail_spare"].labels(device_name).set(smart_log["avail_spare"] / 100)
-                    metrics["spare_thresh"].labels(device_name).set(smart_log["spare_thresh"] / 100)
-                    metrics["percent_used"].labels(device_name).set(smart_log["percent_used"] / 100)
-                    metrics["critical_warning"].labels(device_name).set(
-                        smart_log["critical_warning"]["value"]
-                    )
-                    metrics["media_errors"].labels(device_name).inc(int(smart_log["media_errors"]))
-                    metrics["num_err_log_entries"].labels(device_name).inc(
-                        int(smart_log["num_err_log_entries"])
-                    )
-                    metrics["power_cycles"].labels(device_name).inc(int(smart_log["power_cycles"]))
-                    metrics["power_on_hours"].labels(device_name).inc(
-                        int(smart_log["power_on_hours"])
-                    )
-                    metrics["controller_busy_time"].labels(device_name).inc(
-                        int(smart_log["controller_busy_time"])
-                    )
-                    metrics["unsafe_shutdowns"].labels(device_name).inc(
-                        int(smart_log["unsafe_shutdowns"])
+                    metrics["namespace_info"].labels(
+                        ns_dev,
+                        ns["NSID"],
+                        ctrl_dev,
                     )
 
-                    # NVMe reports temperature in kelvins; convert it to degrees Celsius.
-                    metrics["temperature"].labels(device_name).set(smart_log["temperature"] - 273)
+                    metrics["sector_size"].labels(ns_dev).set(ns["SectorSize"])
+                    metrics["physical_size"].labels(ns_dev).set(ns["PhysicalSize"])
+                    metrics["used_bytes"].labels(ns_dev).set(ns["UsedBytes"])
+
+                # Most SSDs (perhaps _all_ consumer grade SSDs) only contain a single namespace.
```

Comment:
This comment may not be accurate. A vast majority of NVMe SSDs live in data-centers and those may have multiple namespaces. It may make sense to drop the first sentence.